nf_core/utils.py

```diff
@@ -1088,7 +1088,64 @@ def get(self, item: str, default: Any = None) -> Any:
         return getattr(self, item, default)
 
 
-LintConfigType = Optional[Dict[str, Union[List[str], List[Dict[str, List[str]]], bool]]]
+class NFCoreYamlLintConfig(BaseModel):
+    """
+    schema for linting config in `.nf-core.yml` should cover:
+
+    .. code-block:: yaml
+        files_unchanged:
+            - .github/workflows/branch.yml
+        modules_config: False
+        modules_config:
+                - fastqc
+        # merge_markers: False
+        merge_markers:
+                - docs/my_pdf.pdf
+        nextflow_config: False
+        nextflow_config:
+            - manifest.name
+            - config_defaults:
+                - params.annotation_db
+                - params.multiqc_comment_headers
+                - params.custom_table_headers
+        # multiqc_config: False
+        multiqc_config:
+            - report_section_order
+            - report_comment
+        files_exist:
+            - .github/CONTRIBUTING.md
+            - CITATIONS.md
+        template_strings: False
+        template_strings:
+                - docs/my_pdf.pdf
+        nfcore_components: False
+    """
+
+    files_unchanged: Union[bool, List[str]] = []
+    """ List of files that should not be changed """
+    modules_config: Optional[Union[bool, List[str]]] = []
+    """ List of modules that should not be changed """
+    merge_markers: Optional[Union[bool, List[str]]] = []
+    """ List of files that should not contain merge markers """
+    nextflow_config: Optional[Union[bool, List[Union[str, Dict[str, List[str]]]]]] = []
+    """ List of Nextflow config files that should not be changed """
+    multiqc_config: Union[bool, List[str]] = []
+    """ List of MultiQC config options that be changed """
+    files_exist: Union[bool, List[str]] = []
+    """ List of files that can not exist """
```

Comment:
Why not all optional? 
For instance, I can imagine. multiqc_config being absent

Reply:
Optional here, means that it can be of the given type or `None` and because we give them a default value != None, there is no need for that.